test/integration/mount_test.go

```diff
@@ -0,0 +1,115 @@
+// +build integration
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	commonutil "k8s.io/minikube/pkg/util"
+	"k8s.io/minikube/test/integration/util"
+)
+
+func testMounting(t *testing.T) {
+	t.Parallel()
+	minikubeRunner := util.MinikubeRunner{
+		Args:       *args,
+		BinaryPath: *binaryPath,
+		T:          t}
+
+	tempDir, err := ioutil.TempDir("", "mounttest")
+	if err != nil {
+		t.Fatalf("Unexpected error while creating tempDir: %s", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	mountCmd := fmt.Sprintf("mount %s", tempDir)
+	cmd := minikubeRunner.RunDaemon(mountCmd)
+	defer cmd.Process.Kill()
+
+	kubectlRunner := util.NewKubectlRunner(t)
+	podName := "busybox"
+	podPath, _ := filepath.Abs("testdata/busybox-mount-test.yaml")
+
+	// Write file in mounted dir from host
+	expected := "test\n"
+	files := []string{"fromhost", "fromhostremove"}
+	for _, file := range files {
+		path := filepath.Join(tempDir, file)
+		err = ioutil.WriteFile(path, []byte(expected), 0644)
+		if err != nil {
+			t.Fatalf("Unexpected error while writing file %s: %s.", path, err)
+		}
+	}
+	mountTest := func() error {
+		podNamespace := kubectlRunner.CreateRandomNamespace()
+		defer kubectlRunner.DeleteNamespace(podNamespace)
+
+		if _, err := kubectlRunner.RunCommand([]string{"create", "-f", podPath, "--namespace=" + podNamespace}); err != nil {
```

Comment:
--namespace is actually ignored by kubectl. We should probably just remove the namespace handling code for now.